test/success/univalence-types.prl

```diff
@@ -32,28 +32,130 @@ Thm IdIsEquiv{i:lvl} : [
   >
 ].
 
-Thm IdEquiv{i:lvl} : [
-  (-> [ty : (U i hcom)] (Equiv ty ty))
+Thm IdEquiv{l:lvl} : [
+  (-> [ty : (U l hcom)] (Equiv ty ty))
 ] by [
   lam ty.
     unfold Equiv;
     <proj1 = `Id,
-     proj2 = use {IdIsEquiv i} [use ty]>
+     proj2 = use {IdIsEquiv l} [use ty]>
 ].
 
 Def Member(#M; #A) = [
   (= #A #M #M)
 ].
 
-Thm IdUnivalence {i:dim, l:lvl} : [
+Def IdUnivalence {l:lvl,i:dim}(#ty) = [
+  (univalence i #ty #ty ($ {IdEquiv l} #ty))
+].
+
+Thm IdUnivalence/Wf {l:lvl,i:dim} : [
   (->
    [ty : (U l hcom)]
-   (Member
-    (univalence i ty ty ($ {IdEquiv l} ty))
-    (U l)))
+   (Member ({IdUnivalence l i} ty) (U l)))
 ] by [
   lam ty.
     unfold Member;
     refine eq/intro;
     auto
 ].
+
+Thm IdUnivalence/Test0 {l:lvl,i:dim} : [
+  (->
+   [ty : (U l hcom)]
+   [a : ty]
+   (Member (univalence-in i a a) ({IdUnivalence l i} ty)))
+] by [
+  lam ty, a.
+    unfold Member;
+    refine eq/intro;
+    auto
+].
+
+Thm IdUnivalence/Test1 {l:lvl,i:dim} : [
+  (->
+   [ty : (U l hcom)]
+   [a : ty]
+   (= ty (univalence-proj 0 (univalence-in 0 a a) Id) a))
+] by [
+  lam ty, a.
+    refine eq/intro;
+    auto
+].
+
+Def Not = [(lam [b] (if b ff tt))].
+
+Thm Bool/reflect : [
+ (->
+  [a b : bool]
+  [p : (path {_} bool a b)]
+  (= bool a b))
+] by [
+  lam a, b, p. `(coe{0~>1} {x} (= bool ,a (@ ,p x)) ax)
+].
+
+Tac Bool/contra/inverse (#p:exp) = [
+  query gl <- goal.
+  match [gl:jdg] {
+    [a, b | #jdg{%a = %b in bool} =>
+      let eq : [(= bool %b %a)] = use Bool/reflect [`%b, `%a, `#p]; auto. symmetry; elim eq; auto
+    ]
+    [a | %[a:jdg] => id]
+  }
+].
+
+Thm NotIsEquiv : [
+  (IsEquiv bool bool Not)
+] by [
+  unfold IsEquiv; unfold IsContr; unfold Fiber; unfold HasAllPaths;
+  lam b.
+  < proj1 =
+     <proj1 = `($ Not ,b),
```

Comment:
here, you could write `use Not [use b]`.

Reply:
Weird... it raised `Bind`.

Reply:
weird!

Reply:
Is it possible that the `use` tactic relies on knowing the type? `Not` is just a `Def`. In any case, I would like to merge this PR without solving this issue.

Reply:
Oh, you're right!